cloudinit/net/network_manager.py

```diff
@@ -370,8 +371,20 @@ def conn_filename(con_id, target=None):
 
 
 def available(target=None):
-    target_nm_dir = subp.target_path(target, NM_LIB_DIR)
-    return os.path.exists(target_nm_dir)
+    # TODO: Move `uses_systemd` to a more appropriate location
+    # It is imported here to avoid circular import
+    from cloudinit.distros import uses_systemd
+
+    config_present = os.path.isfile(subp.target_path(target, path=NM_CFG_FILE))
+    nmcli_present = subp.which("nmcli", target=target)
+    service_active = True
+    if uses_systemd():
+        try:
+            subp.subp(["systemctl", "is-active", "NetworkManager.service"])
```

Comment:
TLDR: I think we actually want `is-enabled` not `is-active` to assert NetworkManager.service is "turned on". `is-active` means the service is running, not whether it is scheduled to run.

The more I test this, the more I can confirm there is a race in init-local stage with `systemctl is-active NetworkManager.service` because when network is being rendered, the network services have not yet come up fully.  NetworkManager.service is in fact blocked until `After=network-pre.target` but cloud-init-local.service (at least on ubuntu) is `Before=network-pre.target`. So, is-active will return `inactive` in this boot stage and exit non-zero.

I can confirm this behavior when installing network-manager on Ubuntu server and setting renderers: [network-manager, netplan]. `network_manager.available` will always return False in init-local stage and the default renderer chosen remains netplan. Given that both activators and renderers are use the same `network_manager.available` function we'll find ourselves in a state where NM config is rendered, but if activators are triggered, netplan is used to activate the network without any network config written.